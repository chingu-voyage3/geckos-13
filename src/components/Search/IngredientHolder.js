import React from 'react';
import propTypes from 'prop-types';
import { Label, Icon } from 'semantic-ui-react';

// label colors
const colors = ['red', 'orange', 'yellow', 'olive', 'green', 'pink'];

IngredientHolder.propTypes = {
  ingredients: propTypes.array.isRequired,
  onRemoveIngredient: propTypes.func.isRequired,
};

export default function IngredientHolder({ ingredients, onRemoveIngredient }) {
  return (
    <div className="ingredients-holder">
      {ingredients.map((ingredient, i) => (
        <Label
          color={colors[i % colors.length]}
          size="tiny"
          key={i}
          value={ingredient}
        >
          {ingredient}
          <Icon name="delete" onClick={e => onRemoveIngredient(ingredient)} />
        </Label>
      ))}
    </div>
  );
}
