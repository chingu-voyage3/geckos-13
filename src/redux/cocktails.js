export const FETCHING_COCKTAILS = 'FETCHING_COCKTAILS';
export const FETCHING_COCKTAILS_SUCCESS = 'FETCHING_COCKTAILS_SUCCESS';
export const FETCHING_COCKTAILS_FAILURE = 'FETCHING_COCKTAILS_FAILURE';

export const fetchCocktails = () => ({
  type: FETCHING_COCKTAILS,
});

export const fetchCocktailsSuccess = cocktails => ({
  type: FETCHING_COCKTAILS_SUCCESS,
  cocktails,
});

export const fetchCocktailsFailure = err => ({
  type: FETCHING_COCKTAILS_FAILURE,
  err,
});

const initialState = {
  cocktails: [],
};

const cocktails = (state = initialState, action) => {
  switch (action.type) {
    case FETCHING_COCKTAILS:
    case FETCHING_COCKTAILS_SUCCESS:
    case FETCHING_COCKTAILS_FAILURE:
    default:
      return state;
  }
};

export default cocktails;
