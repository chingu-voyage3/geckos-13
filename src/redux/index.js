export { default as cocktails } from './cocktails';
export { default as ingredients } from './ingredients';
