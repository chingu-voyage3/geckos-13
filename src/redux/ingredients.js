export const ADD_INGREDIENT = 'ADD_INGREDIENT';
export const REMOVE_INGREDIENT = 'REMOVE_INGREDIENT';

export const addIngredient = ingredient => ({
  type: ADD_INGREDIENT,
  ingredient,
});

export const removeIngredient = id => ({
  type: REMOVE_INGREDIENT,
  id,
});

const initialState = {
  ingredients: [],
};

const ingredients = (state = initialState, action) => {
  switch (action.type) {
    case ADD_INGREDIENT:
      return {
        ...state,
        ingredients: [...state.ingredients, action.ingredient],
      };
    case REMOVE_INGREDIENT:
      const ingredients = state.ingredients.filter(id => id !== action.id);
      return {
        ...state,
        ingredients,
      };
    default:
      return state;
  }
};

export default ingredients;
