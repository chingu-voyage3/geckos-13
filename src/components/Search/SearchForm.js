import React from 'react';
import Autosuggest from 'react-autosuggest';
import { Form } from 'semantic-ui-react';
import { func } from 'prop-types';

// styles
import './Search.css';

export const ingredientsSuggestions = [
  {
    name: 'Vodka',
    id: 'vodka',
  },
  {
    name: 'Gin',
    id: 'gin',
  },
  {
    name: 'Rum',
    id: 'rum',
  },
  {
    name: 'Ginseng',
    id: 'ginseng',
  },
];

/* calculate suggestions for any given input value.
   return suggestions that start with the same letters as the input value
*/
export const getSuggestions = value => {
  const inputValue = value.trim().toLowerCase();
  const inputLength = inputValue.length;

  return inputLength === 0
    ? []
    : ingredientsSuggestions.filter(
        lang => lang.name.toLowerCase().slice(0, inputLength) === inputValue
      );
};

/* When a suggestion is clicked, populate the input with
   the clicked value
 */
export const getSuggestionValue = suggestion => suggestion.name;

// Use your imagination to render suggestions.
export const renderSuggestion = suggestion => <div>{suggestion.name}</div>;

class SearchForm extends React.Component {
  state = {
    value: '',
    suggestions: [],
  };

  static propTypes = {
    onAddIngredient: func.isRequired,
  };

  onChange = (event, { newValue }) => {
    this.setState({
      value: newValue,
    });
  };

  handleAddIngredient = e => {
    // Only add ingredient if enterkey is pushed
    if (e.charCode !== 13) {
      return;
    }

    this.props.onAddIngredient(this.state.value);
    this.setState({ value: '' });
  };

  // called when suggestions are updated
  onSuggestionsFetchRequested = ({ value }) => {
    this.setState({
      suggestions: getSuggestions(value),
    });
  };

  // call when clearing the suggestions
  onSuggestionsClearRequested = () => {
    this.setState({
      suggestions: [],
    });
  };

  render() {
    const { value, suggestions } = this.state;

    // pass these props to the input.
    const inputProps = {
      placeholder: 'Enter you ingredients and hit enter',
      value,
      onChange: this.onChange,
      onKeyPress: this.handleAddIngredient,
    };

    return (
      <Form size="large">
        <Autosuggest
          suggestions={suggestions}
          onSuggestionsFetchRequested={this.onSuggestionsFetchRequested}
          onSuggestionsClearRequested={this.onSuggestionsClearRequested}
          getSuggestionValue={getSuggestionValue}
          renderSuggestion={renderSuggestion}
          inputProps={inputProps}
        />
      </Form>
    );
  }
}

export default SearchForm;
