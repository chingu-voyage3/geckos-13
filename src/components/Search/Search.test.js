import React from 'react';
import { Label, Icon, Form } from 'semantic-ui-react';

// For testing Search.js
import ConnectedSearch, { SearchContainer } from './SearchContainer';

// For testing SearchForm.js
import SearchForm from './SearchForm';
import {
  getSuggestions,
  ingredientsSuggestions,
  getSuggestionValue,
  renderSuggestion,
} from './SearchForm';
import Autosuggest from 'react-autosuggest';

// For testing IngredientsHolder.js
import IngredientHolder from './IngredientHolder';

// For testing redux
import cocktails from '../../redux/cocktails';
import { combineReducers, createStore, applyMiddleware, compose } from 'redux';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const shallowWithStore = (component, store) => {
  const context = {
    store,
  };
  return shallow(component, { context });
};

describe('---------- Search.js, dumb component ----------', () => {
  it('should match its empty snapshot', () => {
    const component = shallow(<SearchContainer />);

    expect(component).toMatchSnapshot();
  });

  it('should render an h3 with correct text', () => {
    const component = shallow(<SearchContainer />);
    expect(component.find('h3').text()).toBe('Add your ingredients');
  });
});

describe('---------- Search.js connected component --------', () => {
  const initialState = {
    cocktails: [],
    ingredients: [],
  };
  const mockStore = configureStore();
  let store;
  let container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<ConnectedSearch store={store} />);
  });

  it('should create connected testable component', () => {
    expect(container).toBeDefined();
  });

  it('should dispatch ADD_INGREDIENT action', () => {
    expect(container).toBeDefined();
    expect(
      container
        .dive()
        .instance()
        .props.addIngredient('apple')
    ).toEqual({ type: 'ADD_INGREDIENT', ingredient: 'apple' });
  });

  it('should dispatch REMOVE_INGREDIENT action', () => {
    expect(
      container
        .dive()
        .instance()
        .props.removeIngredient('apple')
    ).toEqual({ type: 'REMOVE_INGREDIENT', id: 'apple' });
  });

  //it('should have ingredient array as prop', () => {
  // It works to call the functions on the props, but the cocktails and ingredients arrays are always undefined.
  //console.log(container.dive().instance().props);
  //expect(container.dive().instance().props).toEqual(initialState.ingredients);
  //});

  //it('should have cocktails array as prop', () => {});
});

/*describe('--------- Search.js connected component with provider', () => {
  const initialState = {
    cocktails: [],
    ingredients: [],
  };
  const mockStore = configureStore();
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <ConnectedSearch />
      </Provider>
    );
  });

  it('should create connected testable component', () => {
    expect(wrapper).toBeDefined();
  });
});*/

describe('------------ IngredientHolder.js ----------------', () => {
  const ingredients = ['apple', 'bananas'];
  let onRemoveIngredient = jest.fn();
  it('should match its empty snapshot', () => {
    const component = shallow(
      <IngredientHolder
        ingredients={ingredients}
        onRemoveIngredient={onRemoveIngredient}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render two labels', () => {
    const component = shallow(
      <IngredientHolder
        ingredients={ingredients}
        onRemoveIngredient={onRemoveIngredient}
      />
    );

    const labels = component.find(Label);
    expect(labels.length).toBe(2);
  });

  it('should have props ingredients and onRemoveIngredient"', () => {
    const component = mount(
      <IngredientHolder
        ingredients={ingredients}
        onRemoveIngredient={onRemoveIngredient}
      />
    );

    expect(component.prop('ingredients')).toEqual(ingredients);
    expect(component.prop('onRemoveIngredient')).toBeDefined();
  });

  it('it should have labels with texts "apple" and "bananas""', () => {
    const component = mount(
      <IngredientHolder
        ingredients={ingredients}
        onRemoveIngredient={onRemoveIngredient}
      />
    );

    expect(
      component
        .find(Label)
        .at(0)
        .text()
    ).toBe('apple');

    expect(
      component
        .find(Label)
        .at(1)
        .text()
    ).toBe('bananas');
  });

  it('it should have labels with one Icon"', () => {
    const component = shallow(
      <IngredientHolder
        ingredients={ingredients}
        onRemoveIngredient={onRemoveIngredient}
      />
    );

    expect(
      component
        .find(Label)
        .at(0)
        .find(Icon).length
    ).toBe(1);
  });

  it('it should call onRemoveIngredient when icon is clicked"', () => {
    let onRemoveIngredient = jest.fn();
    const component = shallow(
      <IngredientHolder
        ingredients={ingredients}
        onRemoveIngredient={onRemoveIngredient}
      />
    );

    component
      .find(Label)
      .at(0)
      .find(Icon)
      .simulate('click');

    expect(onRemoveIngredient).toHaveBeenCalled();
  });
});

describe('----------- SearchForm.js --------------', () => {
  const onAddIngredient = jest.fn();
  let component;

  beforeEach(() => {
    component = shallow(<SearchForm onAddIngredient={onAddIngredient} />);
  });

  it('should match its empty snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should return suggestion matching the input', () => {
    expect(getSuggestions('gi')).toEqual([
      { id: 'gin', name: 'Gin' },
      { id: 'ginseng', name: 'Ginseng' },
    ]);
    expect(getSuggestions('v')).toEqual([{ id: 'vodka', name: 'Vodka' }]);
    expect(getSuggestions(' ')).toEqual([]);
  });

  it('should only return the suggestion name', () => {
    expect(getSuggestionValue({ id: 'ginseng', name: 'Ginseng' })).toBe(
      'Ginseng'
    );
  });

  it('should return a div with the suggestion name', () => {
    expect(renderSuggestion({ id: 'ginseng', name: 'Ginseng' })).toEqual(
      <div>Ginseng</div>
    );
  });

  it('should start with the value state as an empty string', () => {
    expect(component.state().value).toBe('');
  });

  it('should render a Form', () => {
    expect(component.find(Form).length).toBe(1);
  });

  it('should render a Autosuggest', () => {
    expect(component.find(Autosuggest).length).toBe(1);
  });
});
