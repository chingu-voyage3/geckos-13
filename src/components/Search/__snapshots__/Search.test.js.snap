// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`---------- Search.js, dumb component ---------- should match its empty snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Add your ingredients
</h3>,
        <SearchForm
          onAddIngredient={[Function]}
/>,
        <IngredientHolder
          ingredients={undefined}
          onRemoveIngredient={[Function]}
/>,
        <Button
          as="button"
          basic={true}
          circular={true}
          fluid={true}
>
          Search
</Button>,
      ],
      "className": "search-container",
      "style": Object {
        "backgroundImage": "url(cocktail-background.jpg)",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add your ingredients",
        },
        "ref": null,
        "rendered": "Add your ingredients",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onAddIngredient": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "ingredients": undefined,
          "onRemoveIngredient": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "button",
          "basic": true,
          "children": "Search",
          "circular": true,
          "fluid": true,
        },
        "ref": null,
        "rendered": "Search",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Add your ingredients
</h3>,
          <SearchForm
            onAddIngredient={[Function]}
/>,
          <IngredientHolder
            ingredients={undefined}
            onRemoveIngredient={[Function]}
/>,
          <Button
            as="button"
            basic={true}
            circular={true}
            fluid={true}
>
            Search
</Button>,
        ],
        "className": "search-container",
        "style": Object {
          "backgroundImage": "url(cocktail-background.jpg)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add your ingredients",
          },
          "ref": null,
          "rendered": "Add your ingredients",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onAddIngredient": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "ingredients": undefined,
            "onRemoveIngredient": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "button",
            "basic": true,
            "children": "Search",
            "circular": true,
            "fluid": true,
          },
          "ref": null,
          "rendered": "Search",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`----------- SearchForm.js -------------- should match its empty snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchForm
    onAddIngredient={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "as": "form",
      "children": <Autosuggest
        alwaysRenderSuggestions={false}
        focusInputOnSuggestionClick={true}
        getSuggestionValue={[Function]}
        highlightFirstSuggestion={false}
        id="1"
        inputProps={
                Object {
                        "onChange": [Function],
                        "onKeyPress": [Function],
                        "placeholder": "Enter you ingredients and hit enter",
                        "value": "",
                      }
        }
        multiSection={false}
        onSuggestionsClearRequested={[Function]}
        onSuggestionsFetchRequested={[Function]}
        renderSuggestion={[Function]}
        renderSuggestionsContainer={[Function]}
        shouldRenderSuggestions={[Function]}
        suggestions={Array []}
        theme={
                Object {
                        "container": "react-autosuggest__container",
                        "containerOpen": "react-autosuggest__container--open",
                        "input": "react-autosuggest__input",
                        "inputFocused": "react-autosuggest__input--focused",
                        "inputOpen": "react-autosuggest__input--open",
                        "sectionContainer": "react-autosuggest__section-container",
                        "sectionContainerFirst": "react-autosuggest__section-container--first",
                        "sectionTitle": "react-autosuggest__section-title",
                        "suggestion": "react-autosuggest__suggestion",
                        "suggestionFirst": "react-autosuggest__suggestion--first",
                        "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
                        "suggestionsContainer": "react-autosuggest__suggestions-container",
                        "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
                        "suggestionsList": "react-autosuggest__suggestions-list",
                      }
        }
/>,
      "size": "large",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alwaysRenderSuggestions": false,
        "focusInputOnSuggestionClick": true,
        "getSuggestionValue": [Function],
        "highlightFirstSuggestion": false,
        "id": "1",
        "inputProps": Object {
          "onChange": [Function],
          "onKeyPress": [Function],
          "placeholder": "Enter you ingredients and hit enter",
          "value": "",
        },
        "multiSection": false,
        "onSuggestionsClearRequested": [Function],
        "onSuggestionsFetchRequested": [Function],
        "renderSuggestion": [Function],
        "renderSuggestionsContainer": [Function],
        "shouldRenderSuggestions": [Function],
        "suggestions": Array [],
        "theme": Object {
          "container": "react-autosuggest__container",
          "containerOpen": "react-autosuggest__container--open",
          "input": "react-autosuggest__input",
          "inputFocused": "react-autosuggest__input--focused",
          "inputOpen": "react-autosuggest__input--open",
          "sectionContainer": "react-autosuggest__section-container",
          "sectionContainerFirst": "react-autosuggest__section-container--first",
          "sectionTitle": "react-autosuggest__section-title",
          "suggestion": "react-autosuggest__suggestion",
          "suggestionFirst": "react-autosuggest__suggestion--first",
          "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
          "suggestionsContainer": "react-autosuggest__suggestions-container",
          "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
          "suggestionsList": "react-autosuggest__suggestions-list",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "as": "form",
        "children": <Autosuggest
          alwaysRenderSuggestions={false}
          focusInputOnSuggestionClick={true}
          getSuggestionValue={[Function]}
          highlightFirstSuggestion={false}
          id="1"
          inputProps={
                    Object {
                              "onChange": [Function],
                              "onKeyPress": [Function],
                              "placeholder": "Enter you ingredients and hit enter",
                              "value": "",
                            }
          }
          multiSection={false}
          onSuggestionsClearRequested={[Function]}
          onSuggestionsFetchRequested={[Function]}
          renderSuggestion={[Function]}
          renderSuggestionsContainer={[Function]}
          shouldRenderSuggestions={[Function]}
          suggestions={Array []}
          theme={
                    Object {
                              "container": "react-autosuggest__container",
                              "containerOpen": "react-autosuggest__container--open",
                              "input": "react-autosuggest__input",
                              "inputFocused": "react-autosuggest__input--focused",
                              "inputOpen": "react-autosuggest__input--open",
                              "sectionContainer": "react-autosuggest__section-container",
                              "sectionContainerFirst": "react-autosuggest__section-container--first",
                              "sectionTitle": "react-autosuggest__section-title",
                              "suggestion": "react-autosuggest__suggestion",
                              "suggestionFirst": "react-autosuggest__suggestion--first",
                              "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
                              "suggestionsContainer": "react-autosuggest__suggestions-container",
                              "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
                              "suggestionsList": "react-autosuggest__suggestions-list",
                            }
          }
/>,
        "size": "large",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alwaysRenderSuggestions": false,
          "focusInputOnSuggestionClick": true,
          "getSuggestionValue": [Function],
          "highlightFirstSuggestion": false,
          "id": "1",
          "inputProps": Object {
            "onChange": [Function],
            "onKeyPress": [Function],
            "placeholder": "Enter you ingredients and hit enter",
            "value": "",
          },
          "multiSection": false,
          "onSuggestionsClearRequested": [Function],
          "onSuggestionsFetchRequested": [Function],
          "renderSuggestion": [Function],
          "renderSuggestionsContainer": [Function],
          "shouldRenderSuggestions": [Function],
          "suggestions": Array [],
          "theme": Object {
            "container": "react-autosuggest__container",
            "containerOpen": "react-autosuggest__container--open",
            "input": "react-autosuggest__input",
            "inputFocused": "react-autosuggest__input--focused",
            "inputOpen": "react-autosuggest__input--open",
            "sectionContainer": "react-autosuggest__section-container",
            "sectionContainerFirst": "react-autosuggest__section-container--first",
            "sectionTitle": "react-autosuggest__section-title",
            "suggestion": "react-autosuggest__suggestion",
            "suggestionFirst": "react-autosuggest__suggestion--first",
            "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
            "suggestionsContainer": "react-autosuggest__suggestions-container",
            "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
            "suggestionsList": "react-autosuggest__suggestions-list",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`------------ IngredientHolder.js ---------------- should match its empty snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IngredientHolder
    ingredients={
        Array [
            "apple",
            "bananas",
          ]
    }
    onRemoveIngredient={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Label
          color="red"
          size="tiny"
          value="apple"
>
          apple
          <Icon
                    as="i"
                    name="delete"
                    onClick={[Function]}
          />
</Label>,
        <Label
          color="orange"
          size="tiny"
          value="bananas"
>
          bananas
          <Icon
                    as="i"
                    name="delete"
                    onClick={[Function]}
          />
</Label>,
      ],
      "className": "ingredients-holder",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "apple",
            <Icon
              as="i"
              name="delete"
              onClick={[Function]}
/>,
          ],
          "color": "red",
          "size": "tiny",
          "value": "apple",
        },
        "ref": null,
        "rendered": Array [
          "apple",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "i",
              "name": "delete",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "bananas",
            <Icon
              as="i"
              name="delete"
              onClick={[Function]}
/>,
          ],
          "color": "orange",
          "size": "tiny",
          "value": "bananas",
        },
        "ref": null,
        "rendered": Array [
          "bananas",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "i",
              "name": "delete",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Label
            color="red"
            size="tiny"
            value="apple"
>
            apple
            <Icon
                        as="i"
                        name="delete"
                        onClick={[Function]}
            />
</Label>,
          <Label
            color="orange"
            size="tiny"
            value="bananas"
>
            bananas
            <Icon
                        as="i"
                        name="delete"
                        onClick={[Function]}
            />
</Label>,
        ],
        "className": "ingredients-holder",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "apple",
              <Icon
                as="i"
                name="delete"
                onClick={[Function]}
/>,
            ],
            "color": "red",
            "size": "tiny",
            "value": "apple",
          },
          "ref": null,
          "rendered": Array [
            "apple",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "i",
                "name": "delete",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "bananas",
              <Icon
                as="i"
                name="delete"
                onClick={[Function]}
/>,
            ],
            "color": "orange",
            "size": "tiny",
            "value": "bananas",
          },
          "ref": null,
          "rendered": Array [
            "bananas",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "i",
                "name": "delete",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
